spec/lib/ruby_require_leak_spec.rb

```diff
@@ -0,0 +1,45 @@
+describe "Ruby 'require' leak with Pathnames" do
+  context "to prevent" do
+    # If this test fails, then either this repo, or one of the provider repos
+    # has added a entry to `config.autoload_paths` that is a Pathname instead
+    # of a raw string.  Until we upgrade to ruby >= 2.5, all Pathnames in the
+    # autoload_paths/$LOAD_PATH should be converted to Strings.
+    #
+    # You can use the code in the expect block as a way of testing this in a
+    # `bin/rails console`:
+    #
+    #     irb> $LOAD_PATH.select { |p| p.class == Pathname }
+    #     #=> []
+    #
+    # More info can be found here:
+    #
+    #     * https://bugs.ruby-lang.org/issues/14372
+    #
+    it "has no Pathnames in the $LOAD_PATH" do
+      expect($LOAD_PATH.select { |p| p.class == Pathname }).to be_empty
+    end
+  end
+
+  context "this file can be removed when" do
+    # If this test fails, then the ManageIQ project has been upgraded to a
+    # version of ruby that no longer requires converting Pathnames in the
+    # autoload_paths/$LOAD_PATH to strings.  This means this file can be
+    # deleted and the changes in the the config/application.rb and
+    # lib/manageiq/**/engine.rb files can be reverted. See the following PRs
+    # for examples of what needs to be reverted:
+    #
+    #     * https://github.com/ManageIQ/manageiq/pull/16837
+    #     * https://github.com/ManageIQ/manageiq-api/pull/288
+    #     * https://github.com/ManageIQ/manageiq-automation_engine/pull/146
+    #     * https://github.com/ManageIQ/manageiq-ui-classic/pull/3266
+    #     * https://github.com/ManageIQ/manageiq-graphql/pull/34
+    #
+    it "is running on ruby >= 2.5.0" do
```

Comment:
I think we want this check even in ruby 2.5.0 since 2.5.0 has a slowdown with these paths.
Also, we'll want to live in 2 versions at the same time (2.4.x and 2.5.x), so this seems to be a stumbling block for that transition in the future.

Maybe have a deprecation warning here instead of a failed test (is this possible?)

Reply:
> Or maybe just merge and cross that road when we get there.

Probably where I am leaning for now, since I don't think we know what the future will hold regarding that.  Also, this is more of a "this code is not going to leak anymore" test, and not checking for performance of requires (in the grand scheme of things, shouldn't really be a performance bottleneck for us anyway).